torch/testing/_internal/optests/autograd_registration.py

```diff
@@ -0,0 +1,135 @@
+import contextlib
+
+import torch
+import torch.utils._pytree as pytree
+
+
+@contextlib.contextmanager
+def set_autograd_fallback_mode(mode):
+    try:
+        prev = torch._C._get_autograd_fallback_mode()
+        torch._C._set_autograd_fallback_mode("nothing")
+        yield
+    finally:
+        torch._C._set_autograd_fallback_mode(prev)
+
+
+def autograd_registration_check(op, args, kwargs):
+    """Check if autograd was registered correctly (for the operator).
+
+    Operators should have "autograd support" registered directly to an
+    autograd dispatch key.
+    An incorrect registration may lead to unexpected silent incorrectness.
+    Note that this check won't catch all problems but will catch
+    the most common ones.
+
+    Example usage:
+        >>> x = torch.randn(3, requires_grad=True)
+        >>> check_autograd_registration(torch.ops.aten.sin.default, (x,), {})
+
+    It is important that you pass us at least one Tensor that has
+    requires_grad=True so that the autograd path gets exercised,
+    otherwise, this check may erroneously pass.
+
+    Here are some best practices if you do find your autograd is
+    registered incorrectly:
+    - If the operator is composite (i.e. consists of other PyTorch ops)
+      and you wish the operator to decompose and get autograd support
+      that way, then please register the implementation to
+      DispatchKey::CompositeImplicitAutograd
+    - If you're adding an autograd formula for the operator, the correct
```

Comment:
Should we recommend the custom op API instead/too?

Reply:
In the future, when the custom op guide is done, I am going to link it from everywhere. I haven't written about the autograd API yet so it wasn't mentioned here